addons/website_sale_comparison/controllers/main.py

```diff
@@ -21,19 +21,21 @@ def product_compare(self, **post):
     @http.route(['/shop/get_product_data'], type='json', auth="public", website=True)
     def get_product_data(self, product_ids, cookies=None):
         ret = {}
-        pricelist_context, pricelist = self._get_pricelist_context()
-        prods = request.env['product.product'].with_context(pricelist_context, display_default_code=False).search([('id', 'in', product_ids)])
+
+        website = request.env['website'].get_current_website()
+        pricelist = website.pricelist_id
+        products = request.env['product.product'].search([('id', 'in', product_ids)])
```

Comment:
Either it is supposed to filter existing records / records on with you have access (and that should be commented), or it should be replaced by a call to ``browse``.

Reply:
Seems legit FYI